cuchem/cuchem/wf/generative/cddd.py

```diff
@@ -92,18 +144,157 @@ def interpolate_smiles(self,
                            scaled_radius=None,
                            force_unique=False,
                            sanitize=True):
-        spec = GenerativeSpec(model=GenerativeModel.MegaMolBART,
-                              smiles=smiles,
-                              radius=scaled_radius,
-                              numRequested=num_points,
-                              forceUnique=force_unique,
-                              sanitize=sanitize)
-
-        result = self.stub.Interpolate(spec)
-        result = list(result.generatedSmiles)
-
-        generated_df = pd.DataFrame({'SMILES': result,
-                                     'Generated': [True for i in range(len(result))]})
-        generated_df.iat[0, 1] = False
-        generated_df.iat[-1, 1] = False
-        return generated_df
+
+        num_points = int(num_points) + 2
+        if len(smiles) < 2:
+            raise Exception('At-least two or more smiles are expected')
+
+        def linear_interpolate_points(embedding, num_points):
+            return np.linspace(embedding[0], embedding[1], num_points)
+
+        result_df = []
+        for idx in range(len(smiles) - 1):
+            data = pd.DataFrame({'transformed_smiles': [smiles[idx], smiles[idx + 1]]})
+            input_embeddings = np.asarray(self.transform(data))
+
+            interp_embeddings = np.apply_along_axis(linear_interpolate_points,
+                                                    axis=0,
+                                                    arr=input_embeddings,
+                                                    num_points=num_points)
+            generated_mols = self.inverse_transform(interp_embeddings, sanitize)
+            interp_embeddings = interp_embeddings.tolist()
+
+            dims = []
+            embeddings = []
+            for interp_embedding in interp_embeddings:
+                dims.append(input_embeddings.shape)
+                interp_embedding = np.asarray(interp_embedding)
+                embeddings.append(interp_embedding)
+
+            interp_df = pd.DataFrame({'SMILES': generated_mols,
+                                      'embeddings': embeddings,
+                                      'embeddings_dim': dims,
+                                      'Generated': [True for i in range(num_points)]})
+
+            # Mark the source and desinations as not generated
+            interp_df.iat[0, 3] = False
+            interp_df.iat[-1, 3] = False
+
+            if force_unique:
+                interp_df = self.compute_unique_smiles(interp_df,
+                                                       self.inverse_transform,
+                                                       scaled_radius=scaled_radius)
+
+            result_df.append(interp_df)
+
+        return pd.concat(result_df)
+
+# import grpc
```

Comment:
What are the comments? If not required remove.

Reply:
CDDD is being removed so the comments will stay to maintain the grpc version